### ISSUE TITLE
使用自己的数据训练PULC 含文字图像方向分类模型模型，保存调用问题

### ISSUE THREAD

**Issue by nainai123** [PaddlePaddle/PaddleClas#3376]
欢迎您使用PaddleClas并反馈相关问题，非常感谢您对PaddleClas的贡献！
提出issue时，辛苦您提供以下信息，方便我们快速定位问题并及时有效地解决您的问题：
 1. PaddleClas版本以及PaddlePaddle版本：请您提供您使用的版本号或分支信息，如PaddleClas release/2.2和PaddlePaddle 2.1.0        paddleclas==2.6.0  paddlepaddle-gpu ==3.0.0
 2. 涉及的其他产品使用的版本号：如您在使用PaddleClas的同时还在使用其他产品，如PaddleServing、PaddleInference等，请您提供其版本号
 3. 训练环境信息：linux 
  a. 具体操作系统，如Linux/Windows/MacOS
  b. Python版本号，如Python3.6/7/8      3.10
  c. CUDA/cuDNN版本， 如CUDA10.2/cuDNN 7.6.5等    11.8
 4. 完整的代码(相比于repo中代码，有改动的地方)、详细的错误信息及相关log
 根据PULC_text_image_orientation.md文件来训练自己的模型
# 全局配置
Global:
  checkpoints: null  # 训练过程中保存的检查点路径，null表示不加载已有检查点
  pretrained_model: null  # 预训练模型路径，null表示不加载预训练模型
  output_dir: ./output/PPcnet_x1_0/  # 输出文件夹路径，用于保存模型、日志等
  device: gpu  # 训练设备，支持gpu或cpu
  save_interval: 20  # 模型保存间隔（以epoch为单位）
  eval_during_train: True  # 是否在训练过程中进行评估
  eval_interval: 10  # 评估间隔（以epoch为单位）
  epochs: 100  # 总训练轮数
  print_batch_step: 10  # 每隔多少个batch打印一次日志
  use_visualdl: False  # 是否使用VisualDL进行可视化
  image_shape: [3, 1700, 1200]  # 输入图像的形状（通道数、高度、宽度）
  save_inference_dir: ./output/PPcnet_x1_1/inference  # 保存推理模型的路径

# 模型架构配置
Arch:
  name: PPLCNet_x1_0  # 使用的模型名称
  class_num: 4  # 分类类别数
  pretrained: True  # 是否加载PaddleClas提供的预训练模型
  use_ssld: True  # 是否使用SSLD（Semi-Supervised Learning Distillation）增强模型性能

# 损失函数配置
Loss:
  Train:
    - CELoss:  # 训练阶段使用交叉熵损失
        weight: 1.0  # 损失权重
        epsilon: 0.1  # Label smoothing参数
  Eval:
    - CELoss:  # 评估阶段使用交叉熵损失
        weight: 1.0  # 损失权重

# 优化器配置
Optimizer:
  name: Momentum  # 使用Momentum优化器
  momentum: 0.9  # 动量参数
  lr:  # 学习率调度策略
    name: Cosine  # 使用余弦退火学习率
    learning_rate: 0.001  # 初始学习率
    warmup_epoch: 5  # 学习率预热的epoch数
  regularizer:
    name: 'L2'  # 使用L2正则化
    coeff: 0.00003  # 正则化系数

# 数据加载器配置
DataLoader:
  Train:  # 训练数据加载器
    dataset:
      name: ImageNetDataset  # 数据集名称
      image_root: /home/yuzai/imageClass/PaddleClas/zlimages_class/  # 图像数据根目录
      cls_label_path: /home/yuzai/imageClass/PaddleClas/zlimages_class/train.txt  # 训练数据标签文件路径
      transform_ops:  # 数据增强操作
        - DecodeImage:  # 解码图像
            to_rgb: True  # 转换为RGB格式
            channel_first: False  # 通道是否放在第一个维度
        - RandCropImage:  # 随机裁剪图像
            size: 224  # 裁剪后的图像大小
        - TimmAutoAugment:  # 自动数据增强
            prob: 0.0  # 应用增强的概率
            config_str: rand-m9-mstd0.5-inc1  # 增强配置
            interpolation: bicubic  # 插值方式
            img_size: 224  # 图像大小
        - NormalizeImage:  # 图像归一化
            scale: 1.0/255.0  # 缩放因子
            mean: [0.485, 0.456, 0.406]  # 均值
            std: [0.229, 0.224, 0.225]  # 标准差
            order: ''  # 通道顺序
        - RandomErasing:  # 随机擦除
            EPSILON: 0.0  # 概率
            sl: 0.02  # 最小擦除面积比例
            sh: 1.0/3.0  # 最大擦除面积比例
            r1: 0.3  # 最小宽高比
            attempt: 10  # 尝试次数
            use_log_aspect: True  # 是否使用对数宽高比
            mode: pixel  # 擦除模式
    sampler:
      name: DistributedBatchSampler  # 分布式采样器
      batch_size: 64  # 每个batch的样本数
      drop_last: False  # 是否丢弃最后一个不完整的batch
      shuffle: True  # 是否打乱数据
    loader:
      num_workers: 8  # 数据加载线程数
      use_shared_memory: True  # 是否使用共享内存

  Eval:  # 评估数据加载器
    dataset:
      name: ImageNetDataset  # 数据集名称
      image_root: /home/yuzai/imageClass/PaddleClas/zlimages_class/  # 图像数据根目录
      cls_label_path: /home/yuzai/imageClass/PaddleClas/zlimages_class/val.txt  # 测试数据标签文件路径
      transform_ops:  # 数据预处理操作
        - DecodeImage:
            to_rgb: True
            channel_first: False
        - ResizeImage:
            resize_short: 256  # 短边调整到256
        - CropImage:
            size: 224  # 中心裁剪到224x224
        - NormalizeImage:
            scale: 1.0/255.0
            mean: [0.485, 0.456, 0.406]
            std: [0.229, 0.224, 0.225]
            order: ''
    sampler:
      name: DistributedBatchSampler
      batch_size: 64
      drop_last: False
      shuffle: False  # 不打乱数据
    loader:
      num_workers: 4
      use_shared_memory: True

# 推理配置
Infer:
  infer_imgs: /home/yuzai/imageClass/PaddleClas/deploy/images/PULC/text_image_orientation/img9_0_270_cw.jpg  # 推理图像路径
  batch_size: 10  # 推理时的batch大小
  transforms:  # 推理时的图像预处理
    - DecodeImage:
        to_rgb: True
        channel_first: False
    - ResizeImage:
        resize_short: 256
    - CropImage:
        size: 224
    - NormalizeImage:
        scale: 1.0/255.0
        mean: [0.485, 0.456, 0.406]
        std: [0.229, 0.224, 0.225]
        order: ''
    - ToCHWImage:  # 转换为CHW格式
  PostProcess:
    name: Topk  # 后处理方法
    topk: 1  # 输出前2个预测结果
    class_id_map_file: ppcls/utils/PULC_label_list/text_image_orientation_label_list.txt  # 类别ID映射文件

# 评估指标配置
Metric:
  Train:
    - TopkAcc:  # 训练阶段的Top-k准确率
        topk: [1, 2]  # 计算Top-1和Top-2准确率
  Eval:
    - TopkAcc:  # 评估阶段的Top-k准确率
        topk: [1, 2]
配置文件如上：
问题1：官网给出的训练结果文件有三个分别为inference.pdiparams、inference.pdiparams.info、inference.pdmodel  我自己根据流程训练完毕只有一个inference.pdiparams（inference.json）
问题2：在使用模型的调用脚本时
from paddleclas import PaddleClas
model = PaddleClas(model_name="text_image_orientation",inference_model_dir="D:/Desktop/model/inference")
result = model.predict(input_data="D:/Desktop/test_img/1.jpg")
print(next(result))   发现没办法正产使用问题1中导出的模型，或者没有能正常调用best_model的脚本可以提供

**Comment by liuhongen1234567** [comment]
您好，paddle3.0 之后套件默认导出的模型更改为json模型，目前paddleclas还没有适配该模型，可以先到paddlex中使用，https://paddlepaddle.github.io/PaddleX/latest/module_usage/tutorials/ocr_modules/textline_orientation_classification.html#_4
使用说明如图：
重新在paddlex导出该模型
    FLAGS_json_format_model=1  python main.py -c /paddle/paddlex_develop1/PaddleX/paddlex/configs/modules/doc_text_orientation/PP-LCNet_x1_0_doc_ori.yaml \
-o Global.mode=export \
-o Export.weight_path='模型动态图路径'

然后将model = create_model(model_name="PP-LCNet_x0_25_textline_ori") 改为 model = create_model(model_name="PP-LCNet_x0_25_textline_ori"，model_dir=XXX)，即可
![Image](https://github.com/user-attachments/assets/ac6c5aa8-92fc-4d87-8652-4387c0f83e89)

**Comment by TingquanGao** [comment]
The issue has no response for a long time and will be closed. You can reopen or new another issue if are still confused.

---
_From Bot_
